### PR TITLE
backend: update from old repo

### PR DESCRIPTION
**Pull request by mads256h**

### CODE REVIEW

**Review by S3JER — APPROVED** [review]

File: backend/src/handlers/devices.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+use axum::{debug_handler, extract::State, http::StatusCode, Json};
+use sqlx::SqlitePool;
+
+use crate::{
+    data_model::device::Device, extractors::auth::Authentication, handlers::util::internal_error,
+    protocol::devices::CreateDeviceRequest,
+};
+
+#[debug_handler]
+pub async fn get_all_smart_devices(
```
Comment (mads256h) [inline backend/src/handlers/devices.rs:10]:
```suggestion
pub async fn get_all_devices(
```

File: backend/src/handlers/devices.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    State(pool): State<SqlitePool>,
+    Authentication(account_id): Authentication,
+) -> Result<Json<Vec<Device>>, (StatusCode, String)> {
+    let devices = sqlx::query!(
+        r#"
+			SELECT id, effect, account_id
+			FROM Devices
+			WHERE account_id = ?
+		"#,
```
Comment (mads256h) [inline backend/src/handlers/devices.rs:19]:
```suggestion
        r#"
	SELECT id, effect, account_id
	FROM Devices
	WHERE account_id = ?
	"#,
```

File: backend/src/handlers/devices.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                id: d.id,
+                effect: d.effect,
+                account_id: d.account_id,
+            })
+            .collect(),
+    ))
+}
+
+#[debug_handler]
+pub async fn create_smart_device(
```
Comment (mads256h) [inline backend/src/handlers/devices.rs:39]:
```suggestion
pub async fn create_device(
```

File: backend/src/handlers/devices.rs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    State(pool): State<SqlitePool>,
+    Authentication(account_id): Authentication,
+    Json(create_device_request): Json<CreateDeviceRequest>,
+) -> Result<Json<Device>, (StatusCode, String)> {
+    let id = sqlx::query_scalar!(
+        r#"
+            INSERT INTO Devices (effect, account_id)
+            VALUES (?, ?)
+            RETURNING id
+        "#,
```
Comment (mads256h) [inline backend/src/handlers/devices.rs:49]:
```suggestion
        r#"
        INSERT INTO Devices (effect, account_id)
        VALUES (?, ?)
        RETURNING id
        "#,
```

File: backend/src/main.rs [hunk 5]
```diff
@@ -41,6 +40,10 @@ fn app(pool: SqlitePool) -> Router {
     Router::new()
         .route("/tasks/all", get(get_tasks))
         .route("/tasks/create", post(create_task))
+        .route("/task/delete", post(delete_task))
+        .route("/device/all", get(get_all_smart_devices))
+        .route("/device/create", post(create_smart_device))
+        .route("/device/delete", post(delete_smart_device))
```
Comment (mads256h) [inline backend/src/main.rs:46]:
```suggestion
        .route("/tasks/delete", post(delete_task))
        .route("/devices/all", get(get_all_smart_devices))
        .route("/devices/create", post(create_smart_device))
        .route("/devices/delete", post(delete_smart_device))
```

File: backend/src/main.rs [hunk 6]
```diff
@@ -112,6 +119,38 @@ mod tests {
...
+            .unwrap();
+
+        let response = ServiceExt::<Request<Body>>::ready(app)
+            .await
+            .unwrap()
+            .call(request)
+            .await
+            .unwrap();
+
+        println!("{:?}", response);
```
Comment (mads256h) [inline backend/src/main.rs:140]: Remove print

**Review by mads256h — COMMENTED** [review]
Small changes

File: backend/src/main.rs (hunk 6 above)
Comment (Darkros1245) [inline backend/src/main.rs:140]:
```suggestion
```

**Review by Darkros1245 — APPROVED** [review]

**Pull request merged**